js/src/components/tree-select-control/index.js

```diff
@@ -169,30 +194,104 @@ const TreeSelectControl = ( {
 			);
 		};
 
-		const filterOption = ( option ) => {
-			if ( hasChildren( option ) ) {
-				option.children = option.children.filter( filterOption );
-				return !! option.children.length;
-			}
+		const descriptors = {
+			allLeaves: {
```

Comment:
💅  I will keep it as `leaves` instead of `allLeaves`. When I saw `all` highlighted in the name of the prop, I was expecting to have just a generic `leaves` somewhere in the code. 

Also the description

`@property {InnerOption[]} allLeaves All (filtered) leaf options under this option.`

I'd suggest to replace it with something like: 

`@property {Leaf[]} leaves All the filtered option children flattened.`

PD. I guess InnerOption is not accurate since we have for example the option.label as well??? 🤔

Reply:
Renamed the `allLeaves` and adjusted some JSDocs in https://github.com/woocommerce/google-listings-and-ads/pull/1522/commits/afa8de0035bd66217caa96b5212fb51ff1ea2be9.

> PD. I guess InnerOption is not accurate since we have for example the option.label as well???

Sorry, I don't quite understand this. The `InnerOption` does have the `label` property. And what does *PD.* refer to? Could you elaborate more?

Reply:
> Sorry, I don't quite understand this. The InnerOption does have the label property. And what does PD. refer to? Could you elaborate more?

Apologies. I missed the label property inside `InnerOption`. All good 👌 (with PD I meant PS, had a typo)